apps/pie-microsite/src/utilities/helpers.js

```diff
@@ -5,14 +5,18 @@
  * @returns {any} The object property value
  */
 const getObjectPropertyByPath = (obj, path) => {
-    const pathSegments = path.split('.');
-    let result = { ...obj };
+    try {
+        const pathSegments = path.split('.');
+        let result = { ...obj };
 
-    for (let i = 0; i < pathSegments.length; i++) {
-        result = result[pathSegments[i]];
-    }
+        for (let i = 0; i < pathSegments.length; i++) {
+            result = result[pathSegments[i]];
+        }
 
-    return result;
+        return result;
+    } catch {
+        return undefined;
```

Comment:
I think it might be worth throwing an error here?

Reply:
Agreed - in general we want the entire site to fail the build if we get errors like this @LTurns , otherwise we risk silent failings getting built. Visual regression _may_ catch them but we should make it as difficult as possible to allow them go through.

Like Faye suggested - throwing a descriptive error here would probably be preferable.

Reply:
@FayeCarter I can see your point, but in this case we actually do want to return undefined so it matches the API you'd expect when querying an object for an unknown property. This means the consume would throw an error when they hit undefined, rather than the helper function which might be used to test for the property's existence.

Reply:
I wish the homer hide emoji was available on github